src/Common/src/Interop/Windows/NtDll/Interop.NtCreateFile.cs

```diff
@@ -59,15 +78,19 @@ internal partial class NtDll
                     ShareAccess: shareAccess,
                     CreateDisposition: createDisposition,
                     CreateOptions: createOptions,
-                    EaBuffer: null,
-                    EaLength: 0);
+                    EaBuffer: eaBuffer,
```

Comment:
You can manually ref count here to keep the other paths allocation free and the generated P/Invoke clean (via DangerousAddRef, DangerousGetHandle, etc).  Here is one example of how to ref count: [https://github.com/dotnet/corefx/blob/e7723074a7ee7cfbc1da516108d2ad85082d77d4/src/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbBuffer.cs#L65](https://github.com/dotnet/corefx/blob/e7723074a7ee7cfbc1da516108d2ad85082d77d4/src/System.Data.Odbc/src/Common/System/Data/ProviderBase/DbBuffer.cs#L65)